File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder ans = new StringBuilder();
        int j = 0;
        for(int i = 0; i < spaces.length; i++) {
            while(j < s.length() && j < spaces[i]) {
                ans.append(s.charAt(j));
                j++;
            }
            
            ans.append(" ");
        }
        
        while(j < s.length()) ans.append(s.charAt(j++));
        return ans.toString();
    }
}

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string ans = "";
        int j = 0;
        for(int i = 0; i < spaces.size(); i++) {
            while(j < spaces[i]) {
                ans += s[j];
                j++;
            }
            
            ans += " ";
        }
        
        while(j < s.length()) ans += s[j++];
        return ans;
    }
};